examples/experimental/dagster-airlift/examples/tutorial-example/tutorial_example_tests/integration_tests/utils.py

```diff
@@ -71,9 +72,18 @@ def poll_for_asset_check(
     evaluation = instance.get_latest_asset_check_evaluation_record(asset_check_key=target)
 
     total_time_elapsed = 0
-    while evaluation is None or evaluation.status not in (
-        AssetCheckExecutionRecordStatus.SUCCEEDED,
-        AssetCheckExecutionRecordStatus.FAILED,
+    while (
+        evaluation is None
+        or evaluation.status
+        not in (
+            AssetCheckExecutionRecordStatus.SUCCEEDED,
+            AssetCheckExecutionRecordStatus.FAILED,
+        )
+        or check.not_none(instance.get_run_by_id(evaluation.run_id)).status
+        not in (
+            DagsterRunStatus.SUCCESS,
+            DagsterRunStatus.FAILURE,
+        )
```

Comment:
Modified this because the tests using it were flaky since this would sometimes return true when the check was completed, but the run was still executing.